projects/backend/controllers/render.ts

```diff
@@ -16,24 +21,75 @@ const fetchRenderedArticle = async (url: string) => {
     }
 }
 
-// TODO: this needs a test once we're happy with the correct format for the paths
-const replaceImageUrls = (html: string): string => {
-    return html.replace(/https:\/\/i.guim.co.uk\/img\//g, `../media/`)
+const fetchCapiArticle = async (
+    internalPageCode: number,
+    capi: CAPIEndpoint,
+): Promise<ArrayBuffer | undefined> => {
+    try {
+        const endpoint = generateCapiEndpoint([internalPageCode], capi)
+        const headers =
+            capi === 'preview' ? await getPreviewHeaders(endpoint) : {}
+        console.log('Making CAPI query', endpoint)
+        console.log('Debug link:', endpoint.replace(/thrift/g, 'json'))
+        const resp = await attempt(fetch(endpoint, { headers }))
+
+        if (hasFailed(resp)) throw new Error('Could not connect to CAPI.')
+
+        if (resp.status != 200) {
+            console.warn(
+                `Non 200 status code: ${resp.status} ${resp.statusText}`,
+            )
+        }
+
+        return await resp.arrayBuffer()
+    } catch (error) {
+        console.warn(
+            `Failed to fetch article from capi (${capi}) for internalPageCode=${internalPageCode}`,
+        )
+        return undefined
+    }
 }
 
 export const renderController = async (req: Request, res: Response) => {
-    const path = req.params.path
-    const renderingUrl = `${process.env.APPS_RENDERING_URL}${path}?editions`
-    console.log(`Fetching ${renderingUrl} from apps rendering`)
-    const renderResponse = await fetchRenderedArticle(renderingUrl)
+    const pageCode = req.params.internalPageCode
+    //The article we are looking for could be in any environment, need to
+    //make an attempt to fetch from all three environment
+    console.log(`Trying to fetch article from capi (live)`)
+    let capiResponse = await fetchCapiArticle(pageCode, 'live')
```

Comment:
The idea here is to try fetch article from one of the capi environment:
It checks if the article is available in `live`, if not then it checks in `printsent`, if not then in `preview`.

Reply:
This makes sense, and seems clearer than any kind of immutable strategy

Reply:
Do you think we should move all this logic to a separate function which encapsulates all the various attempts

```
const fetchLiveCapiArticle = async (internalPageCode: String) =>
	fetchCapiArticle(internalPageCode, 'live');

const fetchPrintSentCapiArticle = async (pageCode: String) => {
	console.log(`Fetching internalPageCode:${pageCode} from printsent CAPI`);
	return fetchCapiArticle(pageCode, 'printsent');
};

const fetchPreviewCapiArticle = async (pageCode: String) => {
	console.log(`Fetching internalPageCode:${pageCode} from preview CAPI`);
	return fetchCapiArticle(pageCode, 'preview');
};

const getCapiResponse = async (pageCode: string): Promise<CapiResponse> => {
	return (
		await fetchLiveCapiArticle(pageCode) ||
		await fetchPrintSentCapiArticle(pageCode) ||
		await fetchPreviewCapiArticle(pageCode) ||
		Promise.reject(
			`Failed to fetch article from capi (print, live & preview) for ${pageCode}`,
		)
	);
};
```

...

const async capiResponse(internalPageCode: string) {
  let capiResponse = await fetchCapiArticle(pageCode, 'live')
  if (capiResponse) {
    return capiResponse
  }
}

Reply:
Thanks @groakland this is a good approach.
I have now recreated a PR where I have taken this feedback

New PR
https://github.com/guardian/editions/pull/1653